src/pages/CustomerInvoicePage.js

```diff
@@ -74,120 +72,50 @@ export const CustomerInvoicePage = ({
   runWithLoadingIndicator,
   routeName,
 }) => {
-  const [tableState, dispatch, instantDebouncedDispatch] = usePageReducer(routeName, {
-    backingData: transaction.items,
-    data: transaction.items.sorted('item.name').slice(),
-    database,
-    keyExtractor,
-    dataState: new Map(),
-    currentFocusedRowKey: null,
-    searchTerm: '',
-    filterDataKeys: ['item.name'],
-    sortBy: 'itemName',
-    isAscending: true,
-  });
-
-  const [modalKey, setModalKey] = useState(null);
-  const [modalIsOpen, setModalIsOpen] = useState(false);
-  const { ITEM_SELECT, COMMENT_EDIT, THEIR_REF_EDIT } = MODAL_KEYS;
-  const { data, dataState, sortBy, isAscending, columns } = tableState;
-  let isSelection = false;
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const row of dataState.values()) {
-    if (row.isSelected) {
-      isSelection = true;
-      break;
-    }
-  }
-
-  const openItemSelector = () => {
-    setModalKey(ITEM_SELECT);
-    setModalIsOpen(true);
-  };
-
-  const openCommentEditor = () => {
-    setModalKey(COMMENT_EDIT);
-    setModalIsOpen(true);
-  };
-
-  const openTheirRefEditor = () => {
-    setModalKey(THEIR_REF_EDIT);
-    setModalIsOpen(true);
-  };
-
-  const closeModal = () => setModalIsOpen(false);
-
-  const getModalTitle = () => {
-    switch (modalKey) {
-      default:
-      case ITEM_SELECT:
-        return modalStrings.search_for_an_item_to_add;
-      case COMMENT_EDIT:
-        return modalStrings.edit_the_invoice_comment;
-      case THEIR_REF_EDIT:
-        return modalStrings.edit_their_reference;
+  const [tableState, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(
```

Comment:
`tableState` should really be `state` now that it has everything and the kitchen sync. Or `pageState`, but state is fine as it usually implies it holds all state for this parent.

Reply:
0e371798027b366c7f70ae9b1271b341d7fb418f